services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -528,21 +544,42 @@ private Set<String> getRegionsToRollForward(
         completedNonTargetRegions.add(nonTargetRegion);
       } else if (version.getStatus().equals(ERROR) || version.getStatus().equals(VersionStatus.KILLED)) {
         failedNonTargetRegions.add(nonTargetRegion);
+      } else if (version.getStatus().equals(ONLINE)) {
+        if (storeResponse.isError()) {
+          String message = "Got error when fetching targetRegionStore: " + storeResponse.getStore();
+          logMessageIfNotRedundant(message);
+          continue;
+        }
+
+        // The in memory store map is out of sync, and we should still allow roll forward to happen
+        // to make the future version current
+        StoreInfo childStore = storeResponse.getStore();
+        if (childStore.getCurrentVersion() < targetVersionNum) {
+          completedNonTargetRegions.add(nonTargetRegion);
+          String message = "Child version " + targetVersionNum + " status is ONLINE for store " + parentStore.getName()
+              + " , but there is still a future version detected in region " + nonTargetRegion
+              + " and the current version is " + parentStore.getCurrentVersion();
```

Comment:
is parent store current version correctly set? we should use child colo current version instead